src/backend/gpopt/translate/CTranslatorDXLToScalar.cpp

```diff
@@ -876,73 +870,20 @@ CTranslatorDXLToScalar::TranslateDXLSubplanTestExprToScalar(
 
 	// Translate arguments
 	List *args = NULL;
+	Expr *outer_expr = NULL;
+	Expr *inner_expr = NULL;
 
 	CDXLNode *outer_child_node = (*test_expr_node)[0];
 	CDXLNode *inner_child_node = (*test_expr_node)[1];
 
-	CContextDXLToPlStmt *dxl_to_plstmt_ctxt =
-		(dynamic_cast<CMappingColIdVarPlStmt *>(colid_var))
-			->GetDXLToPlStmtContext();
-
-	// translate outer expression (can be a deep scalar tree)
-	Expr *outer_arg_expr = NULL;
-	if (has_outer_refs)
-	{
-		Param *param1 = MakeNode(Param);
-		param1->paramkind = PARAM_EXEC;
-
-		CDXLScalarIdent *outer_ident = NULL;
-		Expr *outer_expr = NULL;
-
-		TranslateDXLTestExprScalarIdentToExpr(outer_child_node, param1,
-											  &outer_ident, &outer_expr);
-		GPOS_ASSERT(NULL != outer_ident);
-		GPOS_ASSERT(NULL != outer_expr);
-
-		// finalize outer expression
-		param1->paramtype = CMDIdGPDB::CastMdid(outer_ident->MdidType())->Oid();
-		param1->paramtypmod = outer_ident->TypeModifier();
-		param1->paramid = dxl_to_plstmt_ctxt->GetNextParamId();
-
-		// test expression is used for non-scalar subplan,
-		// first arg of test expression must be an EXEC param1 referring to subplan output
-		args = gpdb::LAppend(args, outer_expr);
+	outer_expr = TranslateDXLToScalar(outer_child_node, colid_var);
```

Comment:
```suggestion
	Expr *outer_expr = TranslateDXLToScalar(outer_child_node, colid_var);
```
Now we assign NULL to the outer_expr and inner_expr variables, and then rewrite them. Do we really need to assign NULL?

Reply:
NULL initialization is optional. Done.